lib/components/joystick/joystick_utils.dart

```diff
@@ -0,0 +1,28 @@
+import 'dart:ui';
+
+import '../../sprite.dart';
+import '../../extensions/offset.dart';
+import '../../extensions/size.dart';
+
+class JoystickUtils {
+  static void renderControl(Canvas c, Sprite sprite, Rect rect, Paint paint) {
```

Comment:
Should this render method maybe also be Vector2 based instead of Rect?

Reply:
I might change that on a followup